packages/typespec-ts/test/modularUnit/type.spec.ts

```diff
@@ -14,8 +14,10 @@ describe("model type", () => {
     assertEqualContent(
       modelFile!.getFullText()!,
       `
+      /** Type of ColorType */
+      export type ColorType = "red" | "blue";
       export interface Test {
-        color: "red" | "blue";
+        color: ColorType;
```

Comment:
talked with the dpg crew about this, for sdk union types like this scenario, where the definition is `, we've decided to just emit it as a string. This is because there isn't information about the name of the union, and dotnet and java can't handle this. If they want an enum, they should define it as an enum in the tsp file, instead of us creating an enum from it

Reply:
That's interesting! But I think emitting this as a string is not proper for TS because we have native support for string union. And for TypeSpec type `"red" | "blue"`, we may prefer to directly map to TypeScript type `"red" | "blue"` like below. But currently we only have the first enum in underlying __raw, it makes us impossible to get all values.

On the other hand for other languages they emit as a string but they may have the requirement to touch following values, e.g for better descriptions or knownable values etc. Just returning the first one would lose these info.

```typescript
export interface Test {
  color: "red" | "blue";
}
```

Reply:
@joheredi Have you been involved in those discussions and are you agree with this ?

Reply:
IIRC we agreed to have it as a string in tcgc since 3/4 languages are just going to emit it as a string, which conflicts with what JS wants here. For JS though, I should be able to fix the `__raw` to return the original union type and there should be enough information there. Down the line, we might decide that we can toggle this behavior with a flag, but since I think it's very JS specific, I think the best way right now is to have JS get the info they need themselves from the raw type

Reply:
Version `0.35.0-dev.4 ` has this change

Reply:
Thank you for the quick fix! Not sure we could have a stable patch version for this?